draco/run.py

```diff
@@ -40,13 +43,25 @@ def run_clingo(
     :param models: Number of models to generate, defaults to 0 (meaning all models).
     :param topK: Whether to return the top K models. If false (default), the program
         will not optimize the output models.
+    :param constants: Dictionary of constants used in genenerator
+        for the maximum number of individual entity types (view, mark and encoding).
+        The default is {"max_views":1, "max_marks":2, "max_encs":4}.
+        You can include a subset of possible keys. Invalid key names will be ignored.
+        If the partial specification contains more entities than in the constants,
+        the generator will refer to the partial specification.
     :yield: The models.
     """
     if not isinstance(program, str):
         program = "\n".join(program)
 
+    flags = []
+    if constants:
+        for key, value in constants.items():
+            if key in ["max_views", "max_marks", "max_encs"]:
```

Comment:
I think we should support any constant ant not just the ones we know about. Imagine if someone wants to run Draco with custom asp, then `run` should still what they use since they cannot modify the Python code when they import the Draco library.